mathlibtools/lib.py

```diff
@@ -70,6 +70,7 @@ class LeanProjectError(Exception):
 
 MATHLIB_URL = 'https://github.com/leanprover-community/mathlib.git'
 LEAN_VERSION_RE = re.compile(r'(.*)\t.*refs/heads/lean-(.*)')
+LEAN_UNESCAPED_IDENTIFIER_RE = re.compile(r"""^(?![λΠΣ])[_a-zA-Zα-ωΑ-Ωϊ-ϻἀ-῾℀-⅏𝒜-𝖟](?:(?![λΠΣ])[_a-zA-Zα-ωΑ-Ωϊ-ϻἀ-῾℀-⅏𝒜-𝖟0-9'ⁿ-₉ₐ-ₜᵢ-ᵪ])*(\\.(?![λΠΣ])[_a-zA-Zα-ωΑ-Ωϊ-ϻἀ-῾℀-⅏𝒜-𝖟](?:(?![λΠΣ])[_a-zA-Zα-ωΑ-Ωϊ-ϻἀ-῾℀-⅏𝒜-𝖟0-9'ⁿ-₉ₐ-ₜᵢ-ᵪ])*)*$""")
```

Comment:
I don't think you want to match dotted names here:
```suggestion
LEAN_UNESCAPED_IDENTIFIER_RE = re.compile(r"""^(?![λΠΣ])[_a-zA-Zα-ωΑ-Ωϊ-ϻἀ-῾℀-⅏𝒜-𝖟](?:(?![λΠΣ])[_a-zA-Zα-ωΑ-Ωϊ-ϻἀ-῾℀-⅏𝒜-𝖟0-9'ⁿ-₉ₐ-ₜᵢ-ᵪ])*$""")
```

Reply:
I tested the dotted names i.e `a.a` using the original regex at https://pythex.org/ and they don't match, which is the right behaviour right? Such filenames would need escaping?

Reply:
You've incorrectly copied the `\\.`, which was originally in a non-escaped string and represented the regex `\.` aka a single `.`. Here it means "a `\` followed by a single character", which is not what you meant.